app/models/course.rb

```diff
@@ -99,13 +99,21 @@ def available_study_modes_from_options
   end
 
   def full?
-    course_options.all?(&:no_vacancies?)
+    course_options.vacancies.blank?
   end
 
   def available?
     course_options.available.present?
   end
 
+  def closed_on_apply?
+    !open_on_apply
+  end
+
+  def not_available?
```

Comment:
Perhaps one for another PR, but this isn't exactly the best method name I've ever picked. Thinking we should be more explicit here and just call it something like `not_available_on_find?`